main.rb

```diff
@@ -0,0 +1,121 @@
+# rubocop: disable Metrics/CyclomaticComplexity, Metrics/MethodLength,Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |n|
+      yield to_a[n]
+    end
+    to_a
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |i|
+      yield to_a[i], i
+    end
+    to_a
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |i| result << i if yield(i) }
+    result
+  end
+
+  def my_all?
+    return true unless block_given?
+
+    result = 0
+    while result < size
+      return false unless yield(self[result])
+
+      result += 1
+    end
+
+    true
+  end
```

Comment:
### #my_all
enumerables my_all? when no block or argument is given returns true when none of the collection members are false or nil
``` ruby
irb>arr = { [1, false, 'hello', [] }
irb>arr.my_all?
  => true
```
> expected to be false
------
Your code isn't receiving arguments when given, check [API DOCK](https://apidock.com/ruby/Enumerable/all%3F)
``` ruby
irb>arr = Array.new(100) { rand(0...9) }
irb>arr.my_all(Integer)?
  => wrong number of arguments (given 1, expected 0)
```
> expected to be true